helix-gateway/src/main/java/org/apache/helix/gateway/participant/HelixGatewayParticipant.java

```diff
@@ -42,17 +43,20 @@
  * for the participant and updates the state of the participant's shards upon successful state
  * transitions signaled by remote participant.
  */
-public class HelixGatewayParticipant {
+public class HelixGatewayParticipant implements HelixManagerStateListener {
   public static final String UNASSIGNED_STATE = "UNASSIGNED";
   private final HelixGatewayServiceProcessor _gatewayServiceProcessor;
   private final HelixManager _participantManager;
+  private final Runnable _onDisconnectedCallback;
   private final Map<String, Map<String, String>> _shardStateMap;
   private final Map<String, CompletableFuture<Boolean>> _stateTransitionResultMap;
 
   private HelixGatewayParticipant(HelixGatewayServiceProcessor gatewayServiceProcessor,
+      Runnable onDisconnectedCallback,
       HelixManager participantManager, Map<String, Map<String, String>> initialShardStateMap) {
     _gatewayServiceProcessor = gatewayServiceProcessor;
     _participantManager = participantManager;
```

Comment:
Not related to this change. May I suggest let's use `helixManager` as var name? Since there is a ParticipantManager class.

Reply:
fixed